ethers/errors.nim

```diff
@@ -1,6 +1,16 @@
 import ./basics
 
-type SolidityError* = object of EthersError
+type
+  SolidityError* = object of EthersError
+  ContractError* = object of EthersError
+  SignerError* = object of EthersError
+  SubscriptionError* = object of EthersError
+  SubscriptionResult*[E] = Result[E, ref SubscriptionError]
```

Comment:
`SubscriptionResult` definitely works... but sometimes it can be a more of a pain than it's worth, mostly when needing to convert between different `Result` types when only the error type is different (eg using some flavour of `mapErr`). You can still return specific error types when it needs inspection at the callsite (since they inherit from CatchableError). See the `reservations` module for examples on how to do this, eg https://github.com/codex-storage/nim-codex/blob/ebee85b1aae05e50839a8f7e911ea888cff209bc/codex/sales/reservations.nim#L273 -- notice the return type of `updateAvailability` is `?!void`, but here we're returning a specific error type). This is the pattern I'd suggest: return/parameter types should `?!T`, then return specific error types where needed.